### PR TITLE
internal/migrate-clarin-dspace5-to-clarin-dspace7-continue

### PR DESCRIPTION
**Pull request by Paurikova2**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  | 0.3  |    0.5 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

## Problem description
Continue of  data migratin from clarin dspace5 to clarin dspace7. Working on bitstream table and tables without API.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/content/ClarinBitstreamServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     * log4j logger
+     */
+    private static Logger log = org.apache.logging.log4j.LogManager.getLogger(ClarinBitstreamServiceImpl.class);
+
+    // Checksum algorithm
+    private static final String CSA = "MD5";
+
+    @Autowired
+    private DSBitStoreService storeService;
+    protected ClarinBitstreamServiceImpl() {
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/content/ClarinBitstreamServiceImpl.java:49]: constructor should be under attributes definition

File: dspace-api/src/main/java/org/dspace/content/ClarinBitstreamServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        Bitstream bitstream = bitstreamDAO.create(context, new Bitstream());
+
+        // Set the format to "unknown"
+        bitstreamService.setFormat(context, bitstream, null);
+        context.addEvent(
+                new Event(Event.CREATE, Constants.BITSTREAM, bitstream.getID(),
+                        null, bitstreamService.getIdentifiers(context, bitstream)));
+
+        //add bitstream to bundle if the bundle is entered
+        if (bundle != null) {
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/content/ClarinBitstreamServiceImpl.java:76]: nonNull

File: dspace-api/src/main/java/org/dspace/content/CommunityServiceImpl.java [hunk 3]
```diff
@@ -115,7 +115,8 @@ public Community create(Community parent, Context context, String handle,
         // of 'anonymous' READ
         Group anonymousGroup = groupService.findByName(context, Group.ANONYMOUS);
 
-        authorizeService.createResourcePolicy(context, newCommunity, anonymousGroup, null, Constants.READ, null);
+        authorizeService.createResourcePolicy(context, newCommunity, anonymousGroup,
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/content/CommunityServiceImpl.java:118]: why

File: dspace-api/src/main/java/org/dspace/content/clarin/ClarinWorkspaceItemServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        return workspaceItem;
+    }
+
+    @Override
+    public WorkspaceItem find(Context context, UUID uuid) throws SQLException {
+        //find workspace item by its UUID
+        WorkspaceItem workspaceItem = workspaceItemDAO.findByID(context, WorkspaceItem.class, uuid);
+
+        //create log if the workspace item is not found
+        if (workspaceItem == null) {
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/content/clarin/ClarinWorkspaceItemServiceImpl.java:62]: Objects.isNull

File: dspace-api/src/main/java/org/dspace/content/service/CollectionService.java [hunk 5]
```diff
@@ -127,6 +127,17 @@ public Collection create(Context context, Community community, String handle, UU
     public Bitstream setLogo(Context context, Collection collection, InputStream is) throws AuthorizeException,
         IOException, SQLException;
 
+    /**
+     * Add the created logo bitstream to collection and create policy to logo bitstream.
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/content/service/CollectionService.java:131]: bitstream logo

File: dspace-api/src/test/java/org/dspace/builder/ClarinBitstreamBuilder.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import java.sql.SQLException;
+
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Bitstream;
+import org.dspace.content.service.DSpaceObjectService;
+import org.dspace.core.Context;
+
+/**
+ * Builder class to build bitstreams in test cases.
+ * This builder was created  for Clarin-Dspace import.
```
Comment (milanmajchrak) [inline dspace-api/src/test/java/org/dspace/builder/ClarinBitstreamBuilder.java:21]: Who created this class

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinBitstreamImportController.java [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+                        && authorizeService.authorizeActionBoolean(context, item, Constants.ADD))) {
+                    throw new AccessDeniedException("You do not have write rights to update the Bundle's item");
+                }
+                if (item != null) {
+                    itemService.update(context, item);
+                }
+                bundleService.update(context, bundle);
+            }
+            bitstreamRest = converter.toRest(bitstream, utils.obtainProjection());
+            context.commit();
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinBitstreamImportController.java:178]: refactor this `if else..`

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLogoImportController.java [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+        //controls
+        if (Objects.isNull(collection) || Objects.isNull(newLogo)) {
+            throw new UnprocessableEntityException(
+                    "The input data are entered incorrectly!");
+        }
+        if (!newLogo.getFormat(context).getShortDescription().equals("Unknown")) {
+            throw new UnprocessableEntityException(
+                    "The bitstream format of bitstream with id: " + newLogo.getID() + " is not Unknown!");
+        }
+        if (collection.getLogo() != null) {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLogoImportController.java:154]: isNotNull

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinEPersonImportController.java [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    /**
+     * Convert String value to Date.
+     * Expects two possible date formats, but more can be added.
+     * @param value
+     * @return converted input value to Date
+     * @throws ParseException if parse error
+     */
+    private Date getDateFromString(String value) throws ParseException {
+        Date output = null;
+        if (StringUtils.isNotBlank(value)) {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinEPersonImportController.java:139]: isBlank do not continue

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinGroupRestController.java [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+     * @throws AuthorizeException if authorization error
+     */
+    @PreAuthorize("hasAuthority('AUTHENTICATED')")
+    @RequestMapping(method = POST, path = "/{uuid}/subgroups")
+    public void addChildGroups(@PathVariable UUID uuid, HttpServletResponse response, HttpServletRequest request)
+            throws SQLException, AuthorizeException {
+
+        Context context = obtainContext(request);
+        Group parentGroup = groupService.find(context, uuid);
+        if (parentGroup == null) {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinGroupRestController.java:81]: isNull

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+        String owningCollectionUuidString = request.getParameter("owningCollection");
+        String multipleTitlesString = request.getParameter("multipleTitles");
+        String publishedBeforeString = request.getParameter("publishedBefore");
+        String multipleFilesString = request.getParameter("multipleFiles");
+        String stageReachedString = request.getParameter("stageReached");
+        String pageReachedString = request.getParameter("pageReached");
+
+        UUID owningCollectionUuid = UUIDUtils.fromString(owningCollectionUuidString);
+        Collection collection = collectionService.find(context, owningCollectionUuid);
+        if (collection == null) {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java:129]: isNull

**Review by milanmajchrak — CHANGES_REQUESTED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinBitstreamImportController.java (hunk 7 above)
Comment (Paurikova2) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinBitstreamImportController.java]: Which part of the code could I refactor?

File: dspace-server-webapp/javac.20230309_154433.args
Comment (mrovnanik) [inline dspace-server-webapp/javac.20230309_154433.args:1]: This cannot remain in the pull-request

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinEPersonImportController.java (hunk 9 above)
Comment (mrovnanik) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinEPersonImportController.java:139]: exactly, stop processing if `value` is blank

File: scripts/export.json.bat
Comment (mrovnanik) [inline scripts/export.json.bat:1]: Why?????

**Review by mrovnanik — CHANGES_REQUESTED** [review]

**Review by mrovnanik — APPROVED** [review]

**Pull request merged**